### PR TITLE
Add Sockets module.

### PR DESCRIPTION
**Pull request by rofinn**
~Currently, based off #544 for testing purposes.~

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -2,6 +2,23 @@ __precompile__()
 
 module Compat
 
+# https://github.com/JuliaLang/julia/pull/25935
+if VERSION < v"0.7.0-DEV.4442"
+    @eval module Sockets
+        import Base:
+            @ip_str, IPAddr, IPv4, IPv6, UDPSocket, TCPSocket,
+            accept, connect, getaddrinfo, getipaddr, getsockname, listen,
```
Comment (ararslan) [inline src/Compat.jl]: Looks like you're missing `bind`.

Comment (rofinn) [reply src/Compat.jl]: Weird. Is there a reason it wasn't deprecated in base? https://github.com/JuliaLang/julia/blob/180111f53155e99fa71bcb461ee562152307720a/base/sysimg.jl#L895

Comment (ararslan) [reply src/Compat.jl]: Indeed. Apparently `Base.bind` is also used for `Channel`s.

Comment (rofinn) [reply src/Compat.jl]: Should I still import and export it in the fake module?

Comment (ararslan) [reply src/Compat.jl]: Might as well, I guess?

**Comment by rofinn** [comment]
Thank you.

**Pull request merged**